monitoring/angle-monitoring/src/main/java/com/farao_community/farao/monitoring/angle_monitoring/AngleMonitoring.java

```diff
@@ -0,0 +1,373 @@
+                || angleCnec.getThresholds().stream()
+                .anyMatch(threshold -> threshold.limitsByMin() && angle != null && angle < threshold.min().orElseThrow());
+    }
+
+    // -------  2) For AngleCnecs with angle overshoot, get associated remedial actions and apply them -----
+    /**
+     * Retrieves the network actions that were defined for an angleCnec (parameter) in a given state (parameter).
+     * Preventive network actions are filtered.
+     */
+    private Set<NetworkAction> getAngleCnecNetworkActions(State state, AngleCnec angleCnec) {
+        Set<RemedialAction<?>> availableRemedialActions =
+                crac.getRemedialActions().stream()
+                        .filter(remedialAction ->
+                                remedialAction.getUsageRules().stream().filter(OnAngleConstraint.class::isInstance)
+                                        .map(OnAngleConstraint.class::cast)
+                                        .anyMatch(onAngleConstraint -> onAngleConstraint.getAngleCnec().equals(angleCnec)))
+                        .collect(Collectors.toSet());
+        if (availableRemedialActions.isEmpty()) {
+            BUSINESS_WARNS.warn("AngleCnec {} in state {} has no associated RA. Angle constraint cannot be secured.", angleCnec.getId(), state.getId());
+            return Collections.emptySet();
+        }
+        if (state.isPreventive()) {
+            BUSINESS_WARNS.warn("AngleCnec {} is constrained in preventive state, it cannot be secured.", angleCnec.getId());
+            return Collections.emptySet();
+        }
+        // Convert remedial actions to network actions
+        return availableRemedialActions.stream().filter(remedialAction -> {
+            if (!(remedialAction instanceof NetworkAction)) {
+                BUSINESS_WARNS.warn("Remedial action {} of AngleCnec {} in state {} is ignored : it's not a network action.", remedialAction.getId(), angleCnec.getId(), state.getId());
+                return false;
+            } else {
+                return true;
+            }
+        }).map(NetworkAction.class::cast).collect(Collectors.toSet());
+    }
+
+    /**
+     * Applies network actions not filtered by checkElementaryActionAndStoreInjection to network
+     * Returns applied network actions.
+     */
+    private Set<NetworkAction> applyNetworkActions(Network networkClone, String angleCnecId, Set<NetworkAction> availableNetworkActions, Set<String> networkElementsToBeExcluded, Map<Country, Double> powerToBeRedispatched) {
+        Set<NetworkAction> appliedNetworkActions = new HashSet<>();
+        boolean networkActionOk = true;
+        for (NetworkAction na : availableNetworkActions) {
+            for (ElementaryAction ea : na.getElementaryActions()) {
+                networkActionOk = networkActionOk && checkElementaryActionAndStoreInjection(ea, networkClone, angleCnecId, na.getId(), networkElementsToBeExcluded, powerToBeRedispatched);
```

Comment:
networkActionOk is true here (until it's not and the second loop is exited right away), so we can just have networkActionOk = checkElementary... instead of networkActionOk = networkActionOk && checkElementary...